src/prefect/storage/bitbucket.py

```diff
@@ -167,3 +252,55 @@ def _get_bitbucket_client(self) -> "atlassian.Bitbucket":
         else:
             session.headers["Authorization"] = "Bearer " + access_token
         return Bitbucket(host, session=session)
+
+    def _get_bitbucket_cloud_client(self) -> "atlassian.Bitbucket":
+        try:
+            from atlassian.bitbucket.cloud import Cloud
+        except ImportError as exc:
+            raise ImportError(
+                "Unable to import atlassian, please ensure you have installed the bitbucket extra"
+            ) from exc
+
+        if self.cloud_username_secret is not None:
+            # If cloud username secret specified, load it
+            cloud_username = Secret(self.cloud_username_secret).get()
+        else:
+            # Otherwise, fallback to loading from local secret or environment variable
+            cloud_username = prefect.context.get("secrets", {}).get(
+                "BITBUCKET_CLOUD_USERNAME"
+            )
+            if cloud_username is None:
+                cloud_username = os.getenv("BITBUCKET_CLOUD_USERNAME")
+
+        if self.cloud_app_password_secret is not None:
+            # If cloud app password secret specified, load it
+            cloud_app_password = Secret(self.cloud_app_password_secret).get()
+        else:
+            # Otherwise, fallback to loading from local secret or environment variable
+            cloud_app_password = prefect.context.get("secrets", {}).get(
+                "BITBUCKET_CLOUD_APP_PASSWORD"
+            )
+            if cloud_app_password is None:
+                cloud_app_password = os.getenv("BITBUCKET_CLOUD_APP_PASSWORD")
+
+        # Bitbucket clould api is rate limited, retry at this frequency
+        # 0.25, 0.5, 1, 2, 4, 8, 16, 32, 64, 128
+        from requests.adapters import HTTPAdapter
+        from requests.packages.urllib3.util.retry import Retry
```

Comment:
These imports should be fine at the top of the module, any particular reason you inlined them?

Reply:
I've moved them to the top.

I was in two minds whether to remove the retry adapter completely.

Reply:
Yeah I'm not sure how I feel about it either. It seems reasonable and a more robust implementation seems like the way to go.

Reply:
It does work. I suggest we keep it. Our Bitbucket Cloud API is under heavy load and it does improve reliability a bit. But of course it is possible all the retries also fail.